flowgate-common/src/main/java/com/vmware/flowgate/common/model/Parent.java

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright 2020 VMware, Inc.
+ * SPDX-License-Identifier: BSD-2-Clause
+*/
+package com.vmware.flowgate.common.model;
+
+public class Parent {
+
+   private ParentType type;
```

Comment:
Lets' use String for the parentType filed. Since there are too much type need to add, and this is a common class we don't want to change the code because just add a new type.